escrow/escrow.go

```diff
@@ -40,12 +42,30 @@ func (h *lockingPaymentChannelService) PaymentChannelFromBlockChain(key *Payment
 	return h.blockchainReader.GetChannelStateFromBlockchain(key)
 }
 
-func (h *lockingPaymentChannelService) PaymentChannel(key *PaymentChannelKey) (channel *PaymentChannelData, ok bool, err error) {
+
+// check whether storage nonce and blockchain nonce of channel are equal or not
+func (h *lockingPaymentChannelService) StorageNonceMatchesWithBlockchainNonce(key *PaymentChannelKey) (equal bool, err error) {
 	storageChannel, storageOk, err := h.storage.Get(key)
 	if err != nil {
 		return
 	}
 
+	blockchainChannel, blockchainOk, err := h.blockchainReader.GetChannelStateFromBlockchain(key)
+	if err != nil {
+		return false, errors.New("channel error:" + err.Error())
+	}
+	if !blockchainOk || !storageOk {
+		return false, errors.New("unable to read channel details.")
```

Comment:
We would know if the error was from block chain or storage no ? , could we add details on where we failed on reading